multiqc/plots/__init__.py

```diff
@@ -0,0 +1,16 @@
+""" HighCharts plots need a globally unique ID embedded in the HTML.
+    - id(pconfig) seems the obvious solution but this might be recycled if
+    a new pconfig object is generated at the same memory address.
+    - random letters work ok but it seems messy to put that code in every chart
+    module.
+    - UUID seems overkill
+    - asking the report for a UID would work but I want to sever the dependency
+    between plots and report classes
+"""
+
+_uid = 0
+
+def get_uid():
```

Comment:
Please use `report.save_htmlid()` instead ([code here](https://github.com/ewels/MultiQC/blob/master/multiqc/utils/report.py#L250-L274)). This collects all IDs across the report to ensure that they're unique without using random strings.

If you want it to work with empty strings then it'd be better to modify that function to return a blank uid in this way. The approach you've used here is nice, but should _really_ be a last resort - I'm trying to avoid using random strings where possible as they are very unhelpful when exporting plots (the unique ID is used there) and in other downstream applications which use plot data (eg. the upcoming MegaQC).

Reply:
Agreed - I think I was actually trying to get rid of the random strings in parallel with your own efforts and just didn't do a thorough job. I'll clean this up.

Reply:
(greps code...) As I suspected - it's imported but never called